src/runtime/components/Modal.vue

```diff
@@ -121,7 +123,11 @@ const ui = computed(() => modal({
     </DialogTrigger>
 
     <DialogPortal :disabled="!portal">
-      <DialogOverlay v-if="overlay" :class="ui.overlay({ class: props.ui?.overlay })" />
+      <DialogOverlay
+        v-if="overlay"
+        :class="ui.overlay({ class: props.ui?.overlay })"
+        @after-leave="emits('after:leave')"
```

Comment:
Should we call this `onClose` instead so its less tied to the `animation` event?

Reply:
Why bind this on the overlay where it can be optional? 🤔 Shouldn't it be better on the DialogContent?